src/main/java/org/isf/medicalstock/manager/MovBrowserManager.java

```diff
@@ -122,4 +151,53 @@ private void check(LocalDateTime from, LocalDateTime to, String errMsgKey) throw
 			}
 		}
 	}
+
+	/**
+	 * Get the last Movement.
+	 *
+	 * @return the retrieved movement.
+	 * @throws OHServiceException
+	 */
+	public Movement getLastMovement() throws OHServiceException {
+		return ioOperations.getLastMovement();
+	}
+
+	/**
+	 * Deletes the last Movement.
+	 *
+	 * @param lastMovement - the last movement to delete
+	 * @throws OHServiceException
+	 */
+	@Transactional
+	public void deleteLastMovement(Movement lastMovement) throws OHServiceException {
+
+		MovementType movType = medicalDsrStockMovTypeManager.getMovementType(lastMovement.getType().getCode());
+		String lotCode = lastMovement.getLot().getCode();
+		Medical medical = lastMovement.getMedical();
+		int medicalCode = medical.getCode();
+		int quantity = lastMovement.getQuantity();
+		if (movType.getType().equals("+")) {
+			medical.setInqty(medical.getInqty() - quantity);
+			medicalsIoOperation.updateMedical(medical);
+			List<Integer> lastMovWithSameLot = ioOperations.getMedicalsFromLot(lotCode);
+			if (lastMovWithSameLot.size() == 1) {
+				lotRepository.deleteById(lotCode);
+			}
+			ioOperations.deleteMovement(lastMovement);
+		} else {
+			Ward ward = lastMovement.getWard();
+			String wardCode = ward.getCode();
+			MedicalWard medWard = movWardBrowserManager.getMedicalWardByWardAndMedical(wardCode, medicalCode, lotCode);
+			medWard.setIn_quantity(medWard.getIn_quantity() - quantity);
+			if (medWard.getIn_quantity() == medWard.getOut_quantity()) {
+				movWardBrowserManager.deleteMedicalWard(medWard);
+			} else {
+				movWardBrowserManager.updateMedicalWard(medWard);
+			}
+			medical.setInqty(medical.getInqty() - quantity);
```

Comment:
Since it is a discharge movement, shouln't you deduct from outgoing quantity?

```suggestion
			medical.setOutqty(medical.getOutqty() - quantity);
```